core/src/main/java/org/kiwix/kiwixmobile/core/main/NightModeViewPainter.kt

```diff
@@ -36,29 +36,32 @@ class NightModeViewPainter @Inject constructor(
   private val invertedPaint =
     Paint().apply { colorFilter = ColorMatrixColorFilter(KiwixWebView.NIGHT_MODE_COLORS) }
 
+  @JvmOverloads
   fun <T : View> update(
     view: T,
     shouldActivateCriteria: ((T) -> Boolean) = { true },
-    vararg additionalViews: View
+    vararg additionalViews: View = emptyArray()
   ) {
     if (nightModeConfig.isNightModeActive()) {
       if (shouldActivateCriteria(view)) {
-        for (view in additionalViews) {
-          activateNightMode(view)
-        }
+        activateNightMode(view, *additionalViews)
       }
     } else {
-      for (view in additionalViews) {
-        deactivateNightMode(view)
-      }
+      deactivateNightMode(view, *additionalViews)
     }
   }
 
-  fun deactivateNightMode(view: View) {
+  fun deactivateNightMode(view: View, vararg additionalViews: View) {
```

Comment:
you don't need the first view parameter. Perhaps a small explanation of varargs is in order

in java you define varags like so `public void myMethod(String... someStrings)` and in the body of the method someStrings is just an array of values. You can invoke that method like so `myMethod("first","second","third")` or by just giving it an array `myMethod(myStringArray)`.

Kotlin works much the same but we have a much more explicit `vararg` as opposed to `...`, the thing that is a bit magical here is the spread operator `*` which (funnily enough) spreads out an array eg
```
val myStringArray = arrayOf("1","2")
myMethod("0", *myStringArray)
```
is equivalent to writing
```
myMethod("0","1","2")
```
but it is important to note that it does copy the entire array, sometimes an issue, not here.

Reply:
@macgills thanks for this information :)
I updated the pr accordingly. Is it fine now?

Reply:
@yashk2000 very very close, see the last few comments

Reply:
@macgills updated. Please take a look.